clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -568,6 +568,128 @@ mlir::LogicalResult CIRToLLVMGlobalOpLowering::matchAndRewrite(
   return mlir::success();
 }
 
+mlir::LogicalResult CIRToLLVMUnaryOpLowering::matchAndRewrite(
+    cir::UnaryOp op, OpAdaptor adaptor,
+    mlir::ConversionPatternRewriter &rewriter) const {
+  assert(op.getType() == op.getInput().getType() &&
+         "Unary operation's operand type and result type are different");
+  mlir::Type type = op.getType();
+  mlir::Type elementType = type;
+  bool isVector = false;
+  assert(!cir::MissingFeatures::vectorType());
+  mlir::Type llvmType = getTypeConverter()->convertType(type);
+  mlir::Location loc = op.getLoc();
+
+  auto createIntConstant = [&](int64_t value) -> mlir::Value {
```

Comment:
Morris just added a `create dummy op` that i wonder if should/could be combined here.

Reply:
The createDummyValue he added is in the CIR codegen, and it's doing something different. These do seem like utility functions that will probably be useful elsewhere though. Or maybe a wrapper function isn't even needed.

Reply:
I just had to create a boolean constant. I don't think this will be the last instance of that, so maybe having utility functions for some fundamental types is a good idea.

Reply:
I'm working on an incubator change to do this. There are lots of opportunities to reuse a utility function there. I'll update this PR once I have a final form for the utility function.

Reply:
On second thought, after looking more at the incubator code, I'm not so sure this is worth having a wrapper. There are a lot of small variations that look the same but aren't. Even in this function in the incubator my change to use these lambdas isn't quite right for all cases. I'm going to put this back closer to the way it was in the incubator. It's very slightly longer, but probably less error prone.